lib/debug/thread_client.rb

```diff
@@ -95,8 +96,12 @@ def << req
       @q_cmd << req
     end
 
+    def generate_info
+      @internal_info = { location: current_frame.location_str, line: current_frame.location.lineno }
+    end
+
     def event! ev, *args
-      @q_evt << [self, @output, ev, *args]
+      @q_evt << [self, @output, ev, @internal_info, *args]
```

Comment:
why don't you call `generate_info` here?

Reply:
Thanks, I'll call it in `event!` method.

Reply:
https://github.com/ruby/debug/pull/48/files#diff-4208807b6db8555c61e42002b3e9c879003c5cfdb32d2b1a285a614b22fdf9d5R103